### PR TITLE
[MinimalWindowManager] Move pointer-drag implementation from examples

### PR DESCRIPTION
**Pull request by AlanGriffiths**
Fixes: #2626

### CODE REVIEW

**Review by Saviq — APPROVED** [review]
Works for me!

File: src/miral/minimal_window_manager.cpp [hunk 1]
```diff
@@ -91,8 +91,12 @@ struct miral::MinimalWindowManager::Impl
     bool handle_touch_event(MirTouchEvent const* event);
 
     void apply_resize_by(Displacement movement);
+
+    static MirInputEventModifier pointer_drag_modifier;
```
Comment (wmww) [inline src/miral/minimal_window_manager.cpp:95]: Why is this static? Why not a property in the window manager instance?

File: include/miral/miral/minimal_window_manager.h [hunk 2]
```diff
@@ -72,6 +73,12 @@ class MinimalWindowManager : public WindowManagementPolicy
     /// Raises newly focused window
     void advise_focus_gained(WindowInfo const& window_info) override;
 
+    /// Change the modifer used to identify a window drag gesture
+    /// The default is mir_input_event_modifier_alt
+    /// \warning this is only expected to be called during initialization, there is no thread synchronisation
+    /// \remark Since MirAL 3.7
+    static void pointer_drag_modifier(MirInputEventModifier new_modifier);
```
Comment (wmww) [inline include/miral/miral/minimal_window_manager.h:80]: I would call this `set_pointer_drag_modifier()` to clarify it's purpose. There should be some way to use this function to disable the behavior but if there is it's not documented.

**Review by wmww — CHANGES_REQUESTED** [review]
It works, and surprisingly it seems to be disabled on the floating window manager (not sure why). I do have a few nits about the approach.

File: include/miral/miral/minimal_window_manager.h (hunk 2 above)
Comment (AlanGriffiths) [reply include/miral/miral/minimal_window_manager.h]: I don't see a need for this to be a mechanism to disable the behaviour, there are many other ways to achieve that.

File: src/miral/minimal_window_manager.cpp (hunk 1 above)
Comment (AlanGriffiths) [reply src/miral/minimal_window_manager.cpp]: Yeah, wasn't the best idea. Have reworked

**Comment by AlanGriffiths** [comment]
> surprisingly it seems to be disabled on the floating window manager

I don't see that, what did you do?

**Comment by wmww** [comment]
Sorry, I think I was being confusing. I meant that alt+drag (correctly) does nothing on the tiling window manager. I suppose that shouldn't be surprising since titlebar drag also doesn't move windows.

**Review by wmww — APPROVED** [review]

**Review by Saviq — APPROVED** [review]
Still works.

bors merge

**Pull request merged**